exploration/quoted-literals.md

```diff
@@ -0,0 +1,390 @@
+# Quoted Literals
+
+Status: **Accepted**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Pull Request</dt>
+		<dd><a href="https://github.com/unicode-org/message-format-wg/pull/477">#477</a></dd>
+	</dl>
+</details>
+
+## Objective
+
+Document the rationale for including quoted literals in MessageFormat
+and for delimiting them with the vertical line character, `|`.
+
+## Background
+
+MessageFormat allows both quoted and unquoted literals.
+Unquoted literals satisfy many common use-cases for literals:
+they are sufficient to represent numbers
+and single-word option values and variant keys.
+Quoted literals are helpful in exotic use-cases.
+
+In early drafts of the MessageFormat syntax,
+quoted literals used to be delimited first with quotation marks (`"foo bar"`),
+and then with round parentheses, e.g. `(foo bar)`.
+See [#263](https://github.com/unicode-org/message-format-wg/issues/263).
+
+[#414](https://github.com/unicode-org/message-format-wg/pull/414) proposed to revert these changes
+and go back to using single and/or double quotes as delimiters.
+The propsal was rejected.
+This document is an artifact of that rejection.
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+In general, quoted literals are useful for:
+
+1. encoding literals containing whitespace, like literals consisting of multiple words,
+1. encoding literals containing exotic characters that do not conform to the `unquoted` production in ABNF.
+
+More specifically:
+
+- Message authors and translators need to be able to use the apostrophe in the message content,
+  and may want to use the single quote character
+  to represent it instead of the typograhic (curly) apostrophe.
+
+  > ```
+  > …{|New Year's Eve|}…
+  > ```
+
+- Message authors may want to use literals to define locale-aware dates as literals in a modified RFC 3339 format:
+
+  > ```
+  > The Unix epoch is defined as {|1970-01-01 00:00:00Z| :datetime}.
+  > ```
+
+- Message authors may want to use multiple words as values of certain options passed to custom functions and markup elements:
+
+  > ```
+  > {+button title=|Click here!|}Submit{-button}
+  > ```
+
+  > [!NOTE]
+  > Quoted literals cannot contain placeholders, making interpolating data into them impossible.
+  >
+  > ```
+  > -- This is impossible in MessageFormat 2.0.
+  > {+button title=|Goodbye, {$userName}!|}Sign out{-button}
+  > ```
+
+- Selector function implementers may want to support multi-word variant keys
+  or exotic characters in variant keys
+  to effectively create "mini-DSLs" for the matching logic:
```

Comment:
```suggestion
- Selector function implementers might need to match different string values
  such as those present in data values.
  These might include keys containing arbitrary text, multiple words,
  or other sequences not otherwise permitted in the syntax.
```